frontend/app/src/components/settings/data-security/OracleCacheManagement.vue

```diff
@@ -134,188 +134,203 @@
+    const toAsset = ref<string>('');
+    const search = ref<string>('');
+    const selection = ref<PriceOracle>('cryptocompare');
+    const deleteEntry = ref<OracleCacheMeta | null>(null);
 
-  get deleteFromAsset(): string {
-    if (this.deleteEntry) {
-      return this.deleteEntry.fromAsset;
-    }
-    return '';
-  }
+    const load = async () => {
+      set(loading, true);
+      set(cacheData, await getPriceCache('cryptocompare'));
+    };
 
-  get deleteToAsset(): string {
-    if (this.deleteEntry) {
-      return this.deleteEntry.toAsset;
-    }
-    return '';
-  }
+    onMounted(async () => {
+      await load();
+    });
 
-  get pending(): boolean {
-    return this.isTaskRunning(TaskType.CREATE_PRICE_CACHE).value;
-  }
+    watch(selection, async () => {
+      await load();
+    });
 
-  async mounted() {
-    await this.load();
-  }
+    const deleteFromAsset = computed<string>(() => {
+      if (get(deleteEntry)) {
+        return get(deleteEntry)!.fromAsset;
+      }
 
-  private async load() {
-    this.loading = true;
-    this.cacheData = await this.$api.balances.getPriceCache('cryptocompare');
-    this.loading = false;
-  }
+      return '';
+    });
 
-  confirmDelete(entry: OracleCacheMeta) {
-    this.confirmClear = true;
-    this.deleteEntry = entry;
-  }
+    const deleteToAsset = computed<string>(() => {
+      if (get(deleteEntry)) {
+        return get(deleteEntry)!.toAsset;
+      }
 
-  async clearCache() {
-    assert(this.deleteEntry);
-    const { fromAsset, toAsset } = this.deleteEntry;
-    this.confirmClear = false;
-    this.deleteEntry = null;
-    try {
-      await this.$api.balances.deletePriceCache(
-        this.selection,
-        fromAsset,
-        toAsset
-      );
-      await this.load();
-    } catch (e: any) {
-      const title = this.$t(
-        'oracle_cache_management.notification.title'
-      ).toString();
+      return '';
+    });
 
-      const message = this.$t('oracle_cache_management.clear_error', {
-        fromAsset,
-        toAsset,
-        error: e.message
-      }).toString();
+    const pending = computed<boolean>(() => {
+      return get(isTaskRunning(TaskType.CREATE_PRICE_CACHE));
+    });
 
-      const { notify } = useNotifications();
-      notify({
-        title,
-        message,
-        severity: Severity.ERROR,
-        display: true
-      });
-    }
-  }
+    const confirmDelete = (entry: OracleCacheMeta) => {
+      set(confirmClear, true);
+      set(deleteEntry, entry);
+    };
 
-  async fetchPrices() {
-    const fromAsset = this.fromAsset;
-    const toAsset = this.toAsset;
-    const source = this.selection;
-    this.fromAsset = '';
-    this.toAsset = '';
+    const { notify } = useNotifications();
 
-    const status = await this.createOracleCache({
-      purgeOld: false,
-      fromAsset: fromAsset,
-      toAsset: toAsset,
-      source: source
-    });
+    const clearCache = async () => {
+      assert(get(deleteEntry));
```

Comment:
also here, we can use  `const blah = get(deleteEntry);` then `assert(blah)` which will make the ! at the end redundant